ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLint.kt

```diff
@@ -93,65 +120,25 @@ object KtLint {
     /**
      * Check source for lint errors.
      *
-     * @param text source
-     * @param ruleSets a collection of "RuleSet"s used to validate source
-     * @param cb callback that is going to be executed for every lint error
-     *
      * @throws ParseException if text is not a valid Kotlin code
      * @throws RuleExecutionException in case of internal failure caused by a bug in rule implementation
      */
-    fun lint(text: String, ruleSets: Iterable<RuleSet>, cb: (e: LintError) -> Unit) {
-        lint(text, ruleSets, emptyMap(), cb, script = false)
-    }
-
-    fun lint(text: String, ruleSets: Iterable<RuleSet>, userData: Map<String, String>, cb: (e: LintError) -> Unit) {
-        lint(text, ruleSets, userData, cb, script = false)
-    }
-
-    /**
-     * Check source for lint errors.
-     *
-     * @param text script source
-     * @param ruleSets a collection of "RuleSet"s used to validate source
-     * @param cb callback that is going to be executed for every lint error
-     *
-     * @throws ParseException if text is not a valid Kotlin code
-     * @throws RuleExecutionException in case of internal failure caused by a bug in rule implementation
-     */
-    fun lintScript(text: String, ruleSets: Iterable<RuleSet>, cb: (e: LintError) -> Unit) {
-        lint(text, ruleSets, emptyMap(), cb, script = true)
-    }
-
-    fun lintScript(
-        text: String,
-        ruleSets: Iterable<RuleSet>,
-        userData: Map<String, String>,
-        cb: (e: LintError) -> Unit
-    ) {
-        lint(text, ruleSets, userData, cb, script = true)
-    }
-
-    private fun lint(
-        text: String,
-        ruleSets: Iterable<RuleSet>,
-        userData: Map<String, String>,
-        cb: (e: LintError) -> Unit,
-        script: Boolean
-    ) {
-        val normalizedText = text.replace("\r\n", "\n").replace("\r", "\n")
+    fun lint(params: Params) {
+        val normalizedText = params.text.replace("\r\n", "\n").replace("\r", "\n")
         val positionByOffset = calculateLineColByOffset(normalizedText)
-        val fileName = if (script) "file.kts" else "file.kt"
-        val psiFile = psiFileFactory.createFileFromText(fileName, KotlinLanguage.INSTANCE, normalizedText) as KtFile
+        val psiFileName = if (params.script) "file.kts" else "file.kt"
+        val psiFile = psiFileFactory.createFileFromText(psiFileName, KotlinLanguage.INSTANCE, normalizedText) as KtFile
         val errorElement = psiFile.findErrorElement()
         if (errorElement != null) {
             val (line, col) = positionByOffset(errorElement.textOffset)
             throw ParseException(line, col, errorElement.errorDescription)
         }
         val rootNode = psiFile.node
-        injectUserData(rootNode, userData)
+        val mergedUserData = params.userData + userDataResolver(params.editorConfigPath, params.debug)(params.fileName)
```

Comment:
Does this mean that editor config overrides userData passed in? If so maybe it's more intuitive to reverse it?

Reply:
Hm, good question. I don't have a strong opinion either way.

Reply:
Reversed it.